File: frontend/src/types/document.ts
export type DocumentStatus = "pending" | "processing" | "done" | "failed";

export interface Receipt {
  amount: number | null;
  currency: string | null;
  merchantName: string | null;
  purchaseDate: string | null;
  category: string | null;
  isManuallyCorrected: boolean;
  amountMatchedBy: string | null;
  dateMatchedBy: string | null;
  merchantMatchedBy: string | null;
  needsManualReview: boolean;
}

export interface ReceiptDocument {
  id: number;
  status: DocumentStatus;
  createdAt: string;
  receipt: Receipt | null;
  lowQualityScan: boolean;
}

export interface DocumentDetail extends ReceiptDocument {
  rawOcrText: string | null;
  imageUrl: string;
  sharpnessScore: number | null;
}

export interface DocumentListResult {
  items: ReceiptDocument[];
  total: number;
  limit: number;
  offset: number;
}

export interface DocumentUpdate {
  amount?: number | null;
  currency?: string | null;
  merchantName?: string | null;
  purchaseDate?: string | null;
  category?: string | null;
}

File: backend/src/backend/schemas/document.py
from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field
from shared.models import DocumentStatus


class ReceiptOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    amount: Decimal | None = None
    currency: str | None = None
    merchant_name: str | None = None
    purchase_date: date | None = None
    category: str | None = None
    is_manually_corrected: bool = False
    amount_matched_by: str | None = None
    date_matched_by: str | None = None
    merchant_matched_by: str | None = None
    needs_manual_review: bool = False


class DocumentOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    status: DocumentStatus
    created_at: datetime
    receipt: ReceiptOut | None = None
    low_quality_scan: bool = False


class DocumentDetailOut(DocumentOut):
    raw_ocr_text: str | None = None
    image_url: str
    sharpness_score: float | None = None


class DocumentListOut(BaseModel):
    items: list[DocumentOut]
    total: int
    limit: int
    offset: int


class DocumentUpdate(BaseModel):
    amount: Decimal | None = Field(default=None, ge=0, lt=Decimal(1000000))
    currency: str | None = Field(default=None, min_length=3, max_length=3)
    merchant_name: str | None = Field(default=None, min_length=1, max_length=255)
    purchase_date: date | None = None
    category: str | None = Field(default=None, min_length=1, max_length=64)
